cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/cube/DefaultCube.java

```diff
@@ -245,9 +245,10 @@ private Aggregation findAggregation(CubeQuery query) {
     return currentBest;
   }
 
-  private Table<Map<String, String>, Long, Long> getTimeSeries(CubeQuery query, FactScanner scanner) {
-    // tag values -> time -> values
-    Table<Map<String, String>, Long, Long> resultTable = HashBasedTable.create();
+  private Map<Map<String, String>, Table<String, Long, Long>> getTimeSeries(CubeQuery query, FactScanner scanner) {
+    // tag values -> { metrics -> time -> values}
```

Comment:
The better description is:

``` java
// {tag values, metric} -> {time -> value}s
```

i.e. {tag values, metric} is the key that identifies a timeseries; while timeseries has collection of time values.

Please fix the comment. Also, maybe it makes sense to use a structure that better denotes the meaning, e.g.:

``` java
Table<Map<String, String>, String, Map<Long, Long>>
```

up to you though - not sure if that will make code more complex...